app/main.py

```
@@ -1,25 +1,74 @@
 from abc import ABC
+from typing import Any, Callable
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Callable, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Callable, owner: Callable) -> Any:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Callable, value: Any) -> None:
+        if value in range(self.min_amount, self.max_amount + 1):
+            setattr(instance, self.protected_name, value)
```

otherwise you need to raise an Exception